src/mirakuru/base.py

```diff
@@ -349,7 +361,11 @@ def process_stopped() -> bool:
         # that it has terminated due to signal `sig`, which is intended. So
         # don't treat that as an error.
         # pylint: disable=invalid-unary-operand-type
-        if exit_code and exit_code != -sig:
+        expected_exit_code = -sig
+        if exp_sig is not None:
+            expected_exit_code = -exp_sig
```

Comment:
On POSIX systems negative result of Popen.returncode means that it's a result of the given signal (signal N results on -N exit code)

Reply:
https://docs.python.org/3/library/subprocess.html#subprocess.Popen.returncode

Reply:
This assumes a cooperating executable being called. In my case, I am running a Click executable with SimpleExecutor. Click, by default, always returns `1` on unhandled signals an in this case, also the `returncode` is then 1. This can easily be verified with:

`executed.py`:

```python
import time

import click


@click.command()
def hello():
    while True:
        time.sleep(10)


if __name__ == "__main__":
    hello()
```

`executor.py`:

```python
from pathlib import Path
import subprocess
import sys
import time
import signal


proc = subprocess.Popen([sys.executable, Path(__file__).parent / "executed.py"])
with proc:
    time.sleep(2)
    print(f"sending {signal.SIGINT}")
    proc.send_signal(signal.SIGINT)

print(f"Returncode: {proc.returncode}")
```

```
❯ python executor.py 
sending 2

Aborted!
Returncode: 1
```

Isn't it a bit unfortunate that a general-purpose process execution capability is assuming full POSIX-compliance here, even though many programs don't provide this?

Reply:
Okay, I understand more now. @languitar could you create an issue so it does not slip me? (or maybe someone else would like to pick up and create a pr?)